mongomock/aggregate.py

```diff
@@ -358,6 +358,15 @@ def _handle_date_operator(self, operator, values):
             return out_value.second
         if operator == '$millisecond':
             return int(out_value.microsecond / 1000)
+        if operator == '$dateToString':
+            if '%L' in out_value['format']:
+                raise NotImplementedError(
+                    'Although %L is a valid date format for the '
+                    '$dateToString operator, it is currently not implemented '
+                    ' in Mongomock.'
+                )
+            return out_value['date'].strftime(out_value['format'])
```

Comment:
What if `date` or `format` parameter is missing?

Reply:
add code to raise OperationFailure